### PR TITLE
feat: add rollback API

### PR DESCRIPTION
**Pull request by andrewrynhard**
This adds an API for rolling back the version of Talos loaded by
the bootloader.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2170)

### CODE REVIEW

File: api/machine/machine.proto [hunk 1]
```diff
@@ -14,6 +14,7 @@ import "common/common.proto";
 
 // The machine service definition.
 service MachineService {
+  rpc Bootstrap(BootstrapRequest) returns (BootstrapResponse);
```
Comment (andrewrynhard) [inline api/machine/machine.proto:17]: ... now I know my ABC's.

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 2]
```diff
@@ -92,6 +93,37 @@ func (s *Server) Reboot(ctx context.Context, in *empty.Empty) (reply *machine.Re
...
+				// NB: We stop the gRPC server since a failed sequence triggers a
+				// reboot.
+				s.server.GracefulStop()
+			}
+		}
+	}()
+
+	reply = &machine.RollbackResponse{
+		Messages: []*machine.Rollback{
+			{},
```
Comment (andrewrynhard) [inline internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:120]: Maybe the response should say which version we rolled back to?

Comment (rsmitty) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:120]: Couldn't hurt.

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 3]
```diff
@@ -92,6 +93,37 @@ func (s *Server) Reboot(ctx context.Context, in *empty.Empty) (reply *machine.Re
...
+// nolint: dupl
+func (s *Server) Rollback(ctx context.Context, in *machine.RollbackRequest) (reply *machine.RollbackResponse, err error) {
+	log.Printf("rollback via API received")
+
+	if err := syslinux.Revert(); err != nil {
+		return nil, fmt.Errorf("failed to revert bootloader: %v", err)
+	}
+
+	go func() {
+		if err := s.Controller.Run(runtime.SequenceReboot, in, runtime.WithForce()); err != nil {
```
Comment (smira) [inline internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:107]: give me my force! :laughing:

**Pull request merged**